src/ApplicationInsights.ServiceFabric.Native/Net45/CodePackageVersionTelemetryInitializer.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace Microsoft.ApplicationInsights.ServiceFabric
+{
+    using System;
+    using System.Fabric;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// A telemetry initializer that will set component version based on the code package version of a Service Fabric Reliable Service.
+    /// </summary>
+    public class CodePackageVersionTelemetryInitializer : ITelemetryInitializer
+    {
+        /// <summary>
+        /// Initializes component version of the telemetry item with the code package version from the Service Fabric runtime activation context.
+        /// </summary>
+        /// <param name="telemetry">The telemetry context to initialize.</param>
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry == null)
+            {
+                throw new ArgumentNullException(nameof(telemetry));
+            }
+
+            if (telemetry.Context?.Component != null && string.IsNullOrEmpty(telemetry.Context.Component.Version))
+            {
+                var activationContext = FabricRuntime.GetActivationContext();
+                telemetry.Context.Component.Version = activationContext.CodePackageVersion;
```

Comment:
Store CodePackageVersion as a member field, so you don't have to call FabricRuntime.GetActivationContext() every time.

Reply:
A good idea, didn't feel like I needed to use ```var buildVersion = LazyInitializer.EnsureInitialized(ref this.version, this.GetVersion);``` as in the the ```BuildInfoConfigComponentVersionTelemetryInitializer ```